sorc/copygb.fd/CMakeLists.txt

```diff
@@ -1,32 +1,21 @@
-
-if(IntelComp)
-  set(fortran_flags "-g" "-O3" "-r8" "-auto" "${OpenMP_Fortran_FLAGS}")
-elseif(GNUComp)
-  set(fortran_flags "-g" "-O3" "-fdefault-real-8" "${OpenMP_Fortran_FLAGS}")
-else()
-  message("unknown compiler!")
-  exit()
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS "-g -r8 -auto")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O3")
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
+  set(CMAKE_Fortran_FLAGS "-g -fdefault-real-8")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O3")
 endif()
 
-set(exe_name copygb)
-
-file(GLOB fortran_src
-  ${CMAKE_CURRENT_SOURCE_DIR}/*.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/*.f90
-  ${CMAKE_CURRENT_SOURCE_DIR}/*.F)
+set(fortran_src copygb.F)
 
+set(exe_name copygb)
 add_executable(${exe_name} ${fortran_src})
 
-target_compile_options(${exe_name} PRIVATE ${fortran_flags})
-set(module_dir "${CMAKE_CURRENT_BINARY_DIR}/include_8")
-set_target_properties(${exe_name} PROPERTIES Fortran_MODULE_DIRECTORY ${module_dir})
-target_include_directories(${exe_name} PUBLIC ${module_dir})
 
-target_link_libraries(${exe_name} bacio_4 w3nco_4 ip_4 sp_4 ${OpenMP_Fortran_FLAGS})
+if(OpenMP_Fortran_FOUND)
```

Comment:
copygb.F has an `ifdef` on [L408](https://github.com/kgerheiser/NCEPLIBS-grib_util/blob/a5b4505e0fe6ac20aa9cce3cff9eae12b672a867/sorc/copygb.fd/copygb.F#L408)
Shouldn't a `compile_definition` be added in the `CMakeLists.txt` to activate this?
```
#ifdef D_OPENMP
      CALL OMP_SET_NUM_THREADS (NTHREADS)
#endif  /* D_OPENMP */
```

Reply:
``D_OPENMP`` is non standard cpp macro. Instead of adding it to ``compile_definition``, I suggest we change the code to use standard OpenMP macro which is ``_OPENMP``.

Reply:
So, if you add `#ifdef _OPENMP` `_OPENMP` is automatically defined if OpenMP is?

Reply:
Either way, a change is needed; in `copygb..F` or `CMakeLists.txt`

Reply:
I added `_OPENMP`

But looking even more closely, `NTHREADS` will never be set to anything but 1 making all this sort of moot.

Reply:
Yes. Every standard compliant OpenMP implementation must define ``_OPENMP``.

https://www.openmp.org/spec-html/5.0/openmpse10.html#x38-450002.2